src/hooks/useReportScrollManager/index.native.ts

```diff
@@ -24,7 +24,9 @@ function useReportScrollManager(): ReportScrollManagerData {
             return;
         }
 
-        flatListRef.current.scrollToOffset({animated: false, offset: 0});
+        setTimeout(() => {
```

Comment:
@esh-g can you please add description why we could not avoid using setTimeout.. did you also try to avoid using it? Just to be 100% sure that we tried everything

Reply:
can we use maybe here   InteractionManager.runAfterInteractions - is should works well with native side - can you try that?

Reply:
@narefyev91 After some more digging, I found the actual root cause! (very sneaky)
We are using a custom `Flatlist` for android and it saves the last scrolled point and goes to it after it comes into focus...
https://github.com/Expensify/App/blob/b1f8bcca9d54b49ab88d2e7f0ba703cf57bee6fd/src/components/FlatList/index.android.js#L42

When the navigation modal is dismissed, this scrolls to the last scrolled point and prevents the `reportScrollManager` from working correctly...

Reply:
good point - i think we can fix that inside specific flatlist for android - can you try to do it?

Reply:
The current logic seems to be that whenever we scroll the list, the offset is saved and when list is refocused, that offset it set again.
Here are some things that I can brainstorm:
1. We can modify the flatlist to make the `scrollPosition` available publicly (possibly with context) instead of using just state and then set the `scrollPosition` to `0` when dismissing modal
https://github.com/Expensify/App/blob/b1f8bcca9d54b49ab88d2e7f0ba703cf57bee6fd/src/components/FlatList/index.android.js#L35 
2. We can somehow prevent the offset from taking place after the focus

But still exploring the options